bot/exts/events/advent_of_code/_cog.py

```diff
@@ -52,13 +60,80 @@ def __init__(self, bot: Bot):
         self.status_task.set_name("AoC Status Countdown")
         self.status_task.add_done_callback(_helpers.background_task_callback)
 
+        self.completer_task.start()
+
+    @tasks.loop(hours=1)
+    async def completer_task(self) -> None:
+        """
+        Every hour check who has completed all 50 stars and give them the completionist role.
+
+        This also removes the role from people who may already have it but have been removed
+        from the leaderboard.
+        """
+        await self.bot.wait_until_guild_available()
+        guild = self.bot.get_guild(Client.guild)
+        completionist_role = guild.get_role(Roles.aoc_completionist)
+
+        aoc_name_to_member_id = {
+            aoc_name: member_id
+            for member_id, aoc_name in await self.account_links.items()
+        }
+
+        try:
+            leaderboard = await _helpers.fetch_leaderboard()
+        except _helpers.FetchingLeaderboardFailedError:
+            await self.bot.send_log("Unable to fetch AoC leaderboard during role sync.")
+            return
+
+        placement_leaderboard = json.loads(leaderboard["placement_leaderboard"])
+        members_who_deserve_role: set[discord.Member] = set()
+
+        for member_aoc_info in placement_leaderboard.values():
+            if not member_aoc_info["stars"] == 50:
+                # Only give the role to people who have completed all 50 stars
+                continue
+
+            member_id = aoc_name_to_member_id[member_aoc_info["name"]]
+            if not member_id:
+                continue
+
+            member = members.get_or_fetch_member(guild, member_id)
```

Comment:
This line is missing an `await`

Reply:
https://github.com/python-discord/sir-lancebot/pull/991/commits/c88abd4bd3bf427f732195aa88a1ca163fa17d2e